src/libexpr/flake/lockfile.cc

```diff
@@ -185,8 +185,15 @@ std::string LockFile::to_string() const
 
 LockFile LockFile::read(const Path & path)
 {
-    if (!pathExists(path)) return LockFile();
-    return LockFile(nlohmann::json::parse(readFile(path)), path);
+    LockFile l;
+    if (!pathExists(path)) l = LockFile();
+    else try {
+        l = LockFile(nlohmann::json::parse(readFile(path)), path);
+    } catch (const nlohmann::json::parse_error & e) {
+        throw Error("the lockfile did not contain valid JSON.");
```

Comment:
The initial error message isn’t very helpful by itself, but it can still give some information as to the origin of the error, so I’d rather keep it.

```suggestion
        e.addTrace(std::nullopt, "while trying to read the flake lockfile");
```